Libraries/Text/RCTTextAttributes.m

```diff
@@ -17,6 +17,17 @@
 
 @implementation RCTTextAttributes
 
+// [TODO(macOS GH#774)
++ (RCTUIColor *)defaultForegroundColor
+{
+  if (@available(iOS 13.0, *)) {
```

Comment:
what does this @available (ios) do in macOS environment?

Reply:
i wonder if it would be much cleaner to read #ifdef macos to do one thing and do special thing for iOS. but thanks for pointing out we already had this for a while so you ain't introducing a new one.

Reply:
I think it all comes down to 'if we ever merge these changes back to rn-core'

Maybe @Saadnajmi  has some opinion here - but I argue w/o a catchAll replacement in other areas I would just continue to stick to the current pattern

Reply:
It looks like this was one of the few changes to iOS code inside rn-macOS. that was never upstreamed to RN Core.

If this is a bug in iOS code (AKA, RN Core) that we want to diff in RN-macOS, I think the workflow is:

1. Make a diff in RN macOS with the appropriate iOS fix. So long as it's within tags and the diff is minimal, I don't think we care too much about how you structure the diff?
2. Engineer who made the diff makes the change upstream in React Native. Generally this won't land till the next version of React Native, so it's. a while away. For this change, the 0.70 branch cut has happened, so it wouldn't land till 0.71
3. Keep the diff in RN-macOS till we do the merge with the upstreamed change. I suppose we could also cherry-pick the change early once it's landed upstream and save ourselves waiting. 

I'm a little confused though. Why does this work in React Native Core (as evidenced by the expo snack in the PR description) but not in React Native macOS. Do our RCTUIColor diffs get in the way?

Reply:
Oh I see, we had some commits that you've linked in the PR description. Fun....